ReSharper.FSharp/src/FSharp.Psi.Features/src/LanguageService/FSharpElementFactory.fs

```diff
@@ -207,7 +212,53 @@ type FSharpElementFactory(languageService: IFSharpLanguageService, psiModule: IP
             addParensIfNeeded rightArg |> ignore
 
             expr
-
+            
+        // TODO: Work out whether these factory methods are the simplest they can be
+        member x.CreateParenPat() =
+            let expr = createLetBinding "(())"
+            let binding = expr.Bindings |> Seq.exactlyOne
+            binding.HeadPattern.As<IParenPat>()
+            
+        member x.CreateTypedPat (pattern: string, typeUsage: ITypeUsage, spaceBeforeColon: bool) : ITypedPat =
+            let preColonSpace = if spaceBeforeColon then " " else ""
```

Comment:
Could you please also add a test case where it'd produce code with both with added space and without it?
This a good example of a test with switching a setting:

https://github.com/saul/fsharp-support/blob/6dfe786ac4c16c946cf1c29de9d3a2b11b7dc673/ReSharper.FSharp/test/src/FSharp.Tests/Daemon/PipeChainTypeHintStageTest.fs#L18-L19